src/main/java/com/example/controllers/FilterController.java

```diff
@@ -0,0 +1,48 @@
+package com.example.controllers;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.repository.CurrencyRepository;
+
+import model.Currency;
+
+@RequestMapping("/filter")
+@RestController
+public class FilterController {
```

Comment:
Is there a need for multiple controllers? I think this should be part of currency controller since it is related.